agent/app/agent.go

```diff
@@ -660,3 +661,15 @@ func mergeTags(localTags []*ecs.Tag, ec2Tags []*ecs.Tag) []*ecs.Tag {
 
 	return utils.MapToTags(tagsMap)
 }
+
+// getHostPublicIPv4AddressFromEC2Metadata will retrieve the PublicIPAddress (IPv4) of this
+// instance through the EC2 API
+func (agent *ecsAgent) getHostPublicIPv4AddressFromEC2Metadata() string {
+	// Get instance ID from ec2 metadata client.
+	hostPublicIPv4Address, err := agent.ec2MetadataClient.PublicIPv4Address()
+	if err != nil {
+		seelog.Criticalf("Unable to retrieve Host Instance PublicIPv4 Address: %v", err)
```

Comment:
I'd recommend having this as Error, not Critical.

Reply:
Fixed!